elyra/pipeline/airflow/processor_airflow.py

```diff
@@ -278,12 +278,34 @@ def _cc_pipeline(self, pipeline, pipeline_name):
                         continue
 
                     # Get corresponding property's value from parsed pipeline
-                    property_value = operation.component_params.get(component_property.ref)
+                    property_value_dict = operation.component_params.get(component_property.ref)
 
+                    # The type and value of this property can vary depending on what the user chooses
+                    # in the pipeline editor. So we get the current active parameter (e.g. StringControl)
+                    # from the activeControl value
+                    active_property_name = property_value_dict['activeControl']
+
+                    # One we have the value (e.g. StringControl) we use can retrieve the value
+                    # assigned to it
+                    property_value = property_value_dict.get(active_property_name, None)
+
+                    # If the value is not found, assign it the default value assigned in parser
+                    if not property_value:
+                        property_value = component_property.value
+
+                    self.log.debug(f"Active property name : {active_property_name}, value : {property_value}")
                     self.log.debug(f"Processing component parameter '{component_property.name}' "
                                    f"of type '{component_property.data_type}'")
 
-                    if component_property.data_type == "string":
+                    if property_value and str(property_value)[0] == '{' and str(property_value)[-1] == '}':
+                        pv = json.loads(json.dumps(property_value))
+                        if isinstance(pv, dict) and set(pv.keys()) == {'value', 'option'}:
+                            parent_node_name = self._get_node_name(sorted_operations, pv['value'])
+                            processed_value = "\"{{ ti.xcom_pull(task_ids='" + parent_node_name + "') }}\""
+                            operation.component_params[component_property.ref] = processed_value
```

Comment:
While working on the tutorial, I found a bug with processing of dictionary types. I entered a dictionary-value into the `data` field of the SimpleHTTPOperator (using the following format `{'key': 'value'}`). That property is then being processed in this block rather than in the `elif component_property.data_type == 'dictionary'` block as it should, and as a result it's getting assigned the value of the control dictionary. 

Here's a screenshot of the produced DAG:

<img width="513" alt="Screen Shot 2021-11-11 at 6 38 57 PM" src="https://user-images.githubusercontent.com/31816267/141388914-5c3551e3-a680-4cae-8e08-6e949615428d.png">

Reply:
fixed! 
# Operator source: {'url': 'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/operators/http_operator.py'}
```
op_379d0a63_873d_4f0e_b1d1_0a59eda6fe7d = SimpleHttpOperator(
    task_id="SimpleHttpOperator",
    endpoint="api/test",
    method="POST",
    data={"test": "test_value"},
    headers={},
    response_check="",
    extra_options={},
    xcom_push=False,
    http_conn_id="http_default",
    log_response=False,
    dag=dag,
)
```

Reply:
also small minor fix for the dict in default values not adding the curly braces